### ISSUE TITLE
`debug:dotenv` does not work with custom `dotenv_path`.

### ISSUE THREAD

**Issue by Bilge** [symfony/symfony#58987]
### Symfony version(s) affected

6.4.15

### Description

`debug:dotenv` does not work with custom `dotenv_path`. Note that Symfony itself scans and interprets all the files correctly, only this console command fails; it does not pick up any files or variables at all.

### How to reproduce

1. Set `extra.runtime.dotenv_path` to a custom path in `composer.json`.
2. Move all dotenv files into the custom path.

### Additional Context

```
❯ bin/console debug:dotenv

Dotenv Variables & Files
========================

Scanned Files (in descending priority)
--------------------------------------

 * ⨯ .env.local.php
 * ⨯ .env.dev.local
 * ⨯ .env.dev
 * ⨯ .env.local
 * ⨯ .env

Variables
---------

 ---------- -------
  Variable   Value
 ---------- -------

 // Note that values might be different between web and CLI.
```

**Comment by Ejamine** [comment]
This command expects the environment variable `SYMFONY_DOTENV_PATH` to be set or it falls back to the project directory. So, if your files are located in a directory called `dotenv`, you can use this command `SYMFONY_DOTENV_PATH=dotenv/.env bin/console debug:dotenv`.
I hope this can help.

**Comment by GromNaN** [comment]
This is a new feature that have been merged in 7.1: https://github.com/symfony/symfony/pull/52638

**Comment by xabbuh** [comment]
@Bilge Can you check if #59007 fixes this?
